src/Features/VisualBasic/Portable/Structure/Providers/MultiLineIfBlockStructureProvider.vb

```diff
@@ -8,15 +9,53 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 
 Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
     Friend Class MultiLineIfBlockStructureProvider
-        Inherits AbstractSyntaxNodeStructureProvider(Of MultiLineIfBlockSyntax)
-
-        Protected Overrides Sub CollectBlockSpans(node As MultiLineIfBlockSyntax,
-                                                  spans As ArrayBuilder(Of BlockSpan),
-                                                  options As OptionSet,
-                                                  cancellationToken As CancellationToken)
-            spans.AddIfNotNull(CreateBlockSpanFromBlock(
-                               node, node.IfStatement, autoCollapse:=False,
-                               type:=BlockTypes.Conditional, isCollapsible:=True))
+        Inherits InternalStructureBlockStructureProvider(Of MultiLineIfBlockSyntax, IfStatementSyntax, ElseIfBlockSyntax, ElseBlockSyntax, EndBlockStatementSyntax)
+
+        Sub New()
+            MyBase.New()
         End Sub
+
+        Friend Overrides Function FullStructuralBlockOutlining(block As MultiLineIfBlockSyntax) As BlockSpan?
+            Return MyBase.FullStructuralBlockOutlining(block, block.IfStatement)
+        End Function
+
+        Friend Overrides Function GetBannerTextOfFullStructuralBlock(block As MultiLineIfBlockSyntax) As IfStatementSyntax
+            Return block.IfStatement
+        End Function
+
+        Friend Overrides Function GetPreambleOutlining(block As MultiLineIfBlockSyntax, cancellationToken As CancellationToken) As BlockSpan?
+            Return If(block Is Nothing OrElse block.IsMissing, Nothing, GetPreambleOutlining(block))
+        End Function
+
+        Friend Overrides Function GetFirstStatementOfPreamble(block As MultiLineIfBlockSyntax) As SyntaxNode
+            Return If(block.Statements.Count > 0, block.Statements(0), Nothing)
+        End Function
+
+        Friend Overrides Function GetInternalStructuralBlocks(block As MultiLineIfBlockSyntax) As SyntaxList(Of ElseIfBlockSyntax)
+            Return block.ElseIfBlocks
+        End Function
+
+        Friend Overrides Function GetBannerTextOfInternalStructuralBlock(InnerBlock As ElseIfBlockSyntax) As String
+            Return InnerBlock.ElseIfStatement.ToString
+        End Function
+
+        Friend Overrides Function GetEpilogueBlockOutlining(block As MultiLineIfBlockSyntax, cancellationToken As CancellationToken) As BlockSpan?
+            With block
```

Comment:
no with blocks in IDE code.

Reply:
It way easier to read with the `with` than without.